### ISSUE TITLE
Improve documentation regarding needed third-parties packages

### ISSUE THREAD

**Issue by gkastrinis** [souffle-lang/souffle#206]
While trying to install souffle on my laptop, I followed the instructions found here https://github.com/souffle-lang/souffle#how-to-compile-and-install but I had to manually install other packages as well that were not documented.

On a linux (Ubuntu 16.04 - 64bit) system:
libtool m4 automake libboost1.58-dev libboost-system1.58-dev libboost-filesystem1.58-dev libboost-thread1.58-dev flex bison libsqlite3-dev

**Comment by HerbertJordan** [comment]
Even with those extra dependencies I can't get it to work on Ubuntu 16.04. When I run `./bootstrap` I keep getting the following error:
```
libtoolize: putting auxiliary files in '.'.
libtoolize: linking file './ltmain.sh'
libtoolize: putting macros in 'm4'.
libtoolize: linking file 'm4/libtool.m4'
libtoolize: linking file 'm4/ltoptions.m4'
libtoolize: linking file 'm4/ltsugar.m4'
libtoolize: linking file 'm4/ltversion.m4'
libtoolize: linking file 'm4/lt~obsolete.m4'
libtoolize: Consider adding 'AC_CONFIG_MACRO_DIRS([m4])' to configure.ac,
libtoolize: and rerunning libtoolize and aclocal.
fatal: No names found, cannot describe anything.
configure.ac:18: error: AC_INIT should be called with package and version arguments
/usr/share/aclocal-1.15/init.m4:29: AM_INIT_AUTOMAKE is expanded from...
configure.ac:18: the top level
autom4te: /usr/bin/m4 failed with exit status: 1
aclocal: error: echo failed with exit status: 1
```
What is the fix for that?

**Comment by gkastrinis** [comment]
I'm not sure. I don't see this in my system. Maybe I had the missing package (if that's the problem) pre-installed before I tried installing souffle.

**Comment by HerbertJordan** [comment]
I have figured out the problem. The auto-conf script is using `git describe` to obtain a version number for the built. However, when I pull from a repo/fork that has not labels attached, this failed, causing the cryptic error above. By adding the flag `--always` this could be prevented.

**Comment by b-scholz** [comment]
I just noticed that libsql3-dev is not part of the debian package. I added it in my recent merge request.

**Comment by b-scholz** [comment]
George - are these issues still there?

**Comment by gkastrinis** [comment]
Maybe I am missing something but I don't see anything changed. The guide on the landing page (https://github.com/souffle-lang/souffle#how-to-compile-and-install) is not mentioning that I need to install those packages and I tried the process in a fresh machine and I still get errors (so it's not the case that missing packages are handled internally by some script or whatever).

E.g.
sh ./bootstrap gives
 ./bootstrap: 6: ./bootstrap: libtoolize: not found
./bootstrap: 9: ./bootstrap: aclocal: not found
./bootstrap: 10: ./bootstrap: autoheader: not found
./bootstrap: 11: ./bootstrap: automake: not found
./bootstrap: 12: ./bootstrap: autoconf: not found

**Comment by limasierra** [comment]
@gkastrinis , in case you still have issues:
- `libtoolize` is provided by package `libtool`
- `aclocal` and `automake` are provided by package `automake`
- `autoheader` and `autoconf` are provided by package `autoconf`

...and you will need the package `sqlite3` as well :)

**Comment by gkastrinis** [comment]
Yes, I know that :) In my original post I also had the packages that needed to be installed

**Comment by mmcgr** [comment]
Fixed in #370. I tested with a fresh Ubuntu 14.04 LTS and the packages listed were sufficient. The wiki also describes the packages needed now.
